File: linguagem c/diagonal_negativos.c
#include <stdio.h>

int main()
{
    int n, i,j, qtdNegativos;

    qtdNegativos = 0;

    printf("Qual é a ordem da matriz?");
    scanf("%d", &n);

    int mat[n][n];

    for(i = 0; i < n; i++){
       for(j = 0; j < n; j++){
          printf("Elemento [%d %d]: ", i, j);
          scanf("%d", &mat[i][j]);

          if(mat[i][j] < 0){
             qtdNegativos = qtdNegativos + 1;
          }
       }
    }

    printf("DIAGONAL PRINCIPAL: \n");
    for(i = 0; i < n; i++){
        printf("%d ",mat[i][i]);
    }

    printf("\nQUANTIDADE NEGATIVOS = %d", qtdNegativos);

}

File: linguagem java/curso_programacao/src/curso_programacao/diagonal_negativos.java
package curso_programacao;

import java.util.Scanner;

public class diagonal_negativos {

	public static void main(String[] args) {
		
		
		Scanner sc  = new Scanner(System.in);
		
		int n, qtdNegativos;
		
		qtdNegativos = 0;
		
		System.out.print("Qual é a ordem da matriz? ");
		n = sc.nextInt();
		
		int[][] mat = new int[n][n];
		
		for(int i = 0; i < n; i++) {
			for(int j = 0; j < n; j++) {
			    System.out.println("Elemento "+ "["+ i+ " , "+ j+ "]:"); 
			    mat[i][j] = sc.nextInt();
			    
			    if(mat[i][j] < 0) {
			    	qtdNegativos++;
			    }
			}
		}
		
		System.out.println("DIAGONAL PRINCIPAL: ");
		for(int i = 0; i < n; i++) {
			System.out.print(mat[i][i]+ " ");
		}
		
		System.out.println();
		System.out.println("QUANTIDADE NEGATIVOS = "+ qtdNegativos);
		
		
		
		sc.close();

	}

}
